lms/static/js/discovery/search_facets_view.js

```diff
@@ -0,0 +1,107 @@
+;(function (define) {
+
+define([
+    'jquery',
+    'underscore',
+    'backbone',
+    'gettext',
+    'js/discovery/facets_view',
+    'js/discovery/facet_view'
+], function ($, _, Backbone, gettext, FacetsView, FacetView) {
+    'use strict';
+
+    return Backbone.View.extend({
+
+        el: '.search-facets',
+
+        tagName: 'div',
+        templateId: '#search_facets_list-tpl',
+        className: 'facets',
+        facetsTypes: {},
+        moreLessLinksTpl: '#more_less_links-tpl',
+
+        events: {
+            'click li': 'addFacet',
+            'click .show-less': 'collapse',
+            'click .show-more': 'expand',
+        },
+
+        initialize: function (facetsTypes) {
+            if(facetsTypes) {
+                this.facetsTypes = facetsTypes;
+            }
+            this.tpl = _.template($(this.templateId).html());
+            this.moreLessTpl = _.template($(this.moreLessLinksTpl).html());
+            this.$el.html(this.tpl());
+            this.facetViews = [];
+            this.$facetViewsEl = this.$el.find('.search-facets-lists');
+        },
+
+        render: function () {
+            return this;
+        },
+
+        collapse: function(event) {
+            var $el = $(event.currentTarget),
+                $more = $el.siblings('.show-more'),
+                $ul = $el.parent('div').siblings('ul');
+
+            event.preventDefault();
+
+            $ul.css('max-height', '').addClass('collapse');
+            $el.addClass('hidden');
+            $more.removeClass('hidden');
+        },
+
+        expand: function(event) {
+            var $el = $(event.currentTarget),
+                $ul = $el.parent('div').siblings('ul'),
+                facets = $ul.find('li').length,
+                itemHeight = 34;
+
+            event.preventDefault();
+
+            $el.addClass('hidden');
+            $ul.removeClass('collapse').css('max-height', facets * itemHeight + 'px');
```

Comment:
I'm not sure why we need to manipulate the `max-height` CSS property directly here.  Can we set up the CSS styles so the `<ul>` expands automatically to accommodate its children?